xcode/Subconscious/Shared/Parsers/Subtext.swift

```diff
@@ -550,16 +550,22 @@ extension Subtext {
 
 extension Subtext {
     /// Derive an excerpt
-    func excerpt() -> String {
+    func excerpt(fallback: String = "") -> String {
         for block in blocks {
             switch block {
             case .empty:
                 continue
             default:
-                return String(block.body())
+                let block = String(block.body())
+                return block.isEmpty ? fallback : block
             }
         }
-        return ""
+        return fallback
+    }
+    
+    static func excerpt(markup: String, fallback: String = "") -> String {
```

Comment:
Cheaper function that derives excerpt from 512 char substring, so we don't have to parse a large subtext document.